Source/DafnyRuntime/DafnyRuntimeRust/src/lib.rs

```diff
@@ -3774,54 +3899,20 @@ pub mod rcmut {
         mem::transmute(this.get())
     }
 
-    pub unsafe fn downcast<T: 'static>(this: RcMut<dyn ::std::any::Any>) -> Option<RcMut<T>> {
-        let t: Rc<dyn ::std::any::Any> = to_rc(this);
+    #[cfg(feature = "sync")]
+    pub unsafe fn downcast<T: 'static + Send + Sync>(
+        this: RcMut<crate::DynAny>,
+    ) -> Option<RcMut<T>> {
+        let t: Rc<crate::DynAny> = to_rc(this);
         let t: Rc<T> = Rc::downcast::<T>(t).ok()?;
         mem::transmute(t)
     }
 
-    /// A reference counted smart pointer with unrestricted mutability.
-    pub struct ArcMut<T: ?Sized> {
```

Comment:
Now I'm curious, why was this already in here?

Reply:
Because I had envisioned since the beginning we might want to have the async version of Rc for objects; however, I had not used it. So removing it because now for sure we are implementing it differently.